test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt

```diff
@@ -309,3 +324,59 @@ private suspend fun execute(
         fail("Unexpected error during engine execution", e)
     }
 }
+
+fun validate(
+    fixture: TestFixture,
+    hook: EngineTestHook,
+) {
+    val nadel = Nadel.newNadel()
+        .engineFactory {
+            object : NadelExecutionEngine {
+                override fun execute(
+                    executionInput: ExecutionInput,
+                    queryDocument: Document,
+                    instrumentationState: InstrumentationState?,
+                    nadelExecutionParams: NadelExecutionParams,
+                ): CompletableFuture<ExecutionResult> {
+                    throw UnsupportedOperationException("no-op")
```

Comment:
I mentioned this to @felipe-gdr but in the future I think we should split the creation of the "engine schema" and "services" from the `Nadel` object e.g.

1. Introduce new `NadelSchemaBuilder`
2. Create engine schema and services using `NadelSchemaBuilder` say the output is some `NadelSchemas` object
3. Make Nadel.Builder take in `NadelSchemas` _or_ optionally work as today, but delegate off to `NadelSchemaBuilder`

The issue today is creating a Nadel object is very closely tied to creating a `NextgenEngine` which then creates a blueprint, and the blueprint assumes a valid schema. So for validation, to get the engine schema and services out of the Nadel object we need to stub the engine so no blueprint is instantiated.